File: task-03/prime-numbers.cpp
#include <iostream>
using namespace std;

bool isPrime(int num) {
    if (num <= 1) return false;
    for (int i = 2; i * i <= num; ++i)
        if (num % i == 0) return false;
    return true;
}

int main() {
    int upperLimit;
    cout << "Enter an upper limit: ";
    cin >> upperLimit;

    cout << "Prime numbers between 2 and " << upperLimit << " are: ";
    for (int num = 2; num <= upperLimit; ++num)
        if (isPrime(num)) cout << num << " ";
    
    return 0;
}

File: task-03/prime_numbers.rb
def is_prime(n)
  return false if n <= 1

  (2..Math.sqrt(n)).each do |i|
    return false if n % i == 0
  end

  true
end

def find_primes_up_to(limit)
  primes = []
  (2..limit).each do |num|
    primes << num if is_prime(num)
  end
  primes
end

puts "Enter a number: "
user_input = gets.chomp.to_i

if user_input < 2
  puts "There are no prime numbers less than 2."
else
  prime_numbers = find_primes_up_to(user_input)
  puts "Prime numbers from 2 to #{user_input}: #{prime_numbers.join(', ')}"
end
